sharpy/generators/polaraeroforces.py

```diff
@@ -263,16 +268,40 @@ def correct_surface_area_in_case_of_shared_surfaces(self, inode, struct2aero_map
         Returns:
             float: corrected surface area of other surfaces
         '''
-        area = 0.
-        n_surfaces_shared_by_node = len(struct2aero_mapping[inode])
-        if n_surfaces_shared_by_node > 1:
+        if self.flag_shared_node_by_surfaces:
+            n_surfaces_shared_by_node = len(struct2aero_mapping[inode])
             # add area for all other surfaces connected to this node
             for isurf in range(1,n_surfaces_shared_by_node):
                 shared_surf = struct2aero_mapping[inode][isurf]['i_surf']
                 i_n_shared_surf = struct2aero_mapping[inode][isurf]['i_n']
                 _, span_shared_surf, _, chord_shared_surf = span_chord(i_n_shared_surf, zeta_ts[shared_surf])
                 area += span_shared_surf * chord_shared_surf
+        elif self.flag_wingtip_node:
+            area *= 2.
         return area
+        
+    def check_for_special_cases(self, aerogrid):
+        '''
+        Checks if the outboard node is shared by multiple surfaces or single wingtip panel. 
+
+        Args:
+            aerogrid :class:`~sharpy.aero.models.AerogridLoader
+        '''
+        # check if outboard node of aerosurface
+        self.flag_shared_node_by_surfaces, self.flag_wingtip_node = np.zeros((self.n_node,1)), np.zeros((self.n_node,1))
+        for inode in range(self.n_node):
+            if aerogrid.aero_dict['aero_node'][inode]:
+                i_n = aerogrid.struct2aero_mapping[inode][0]['i_n']                
+                isurf = aerogrid.struct2aero_mapping[inode][0]['i_surf']
+                N = aerogrid.aero_dimensions[isurf, 1]
+                if i_n in [0, N]:
+                    if len(aerogrid.struct2aero_mapping[inode]) > 1:
+                        self.flag_shared_node_by_surfaces = 1
```

Comment:
... but here they are defined as int, [inode] missing? If so check the use of the flags in other places